resources/js/recovery/Recovery.vue

```diff
@@ -0,0 +1,377 @@
+                    </div>
+                </div>
+            </div>
+            <div class="flex gap-2 self-end flex-wrap">
+
+                <button class="btn2 btn-default btn-sm" v-if="!hasSelection()" @click="selectAll">
+                    <i class="fa-solid fa-list-check" aria-hidden="true" />
+                    <span v-html="trans('select_all')"></span>
+                </button>
+                <button class="btn2 btn-default btn-sm" v-if="hasSelection()" @click="deselectAll">
+                    <i class="fa-solid fa-xmark" aria-hidden="true" />
+                    <span v-html="trans('deselect_all')"></span>
+                </button>
+                <button class="btn2 btn-primary btn-sm" v-if="hasSelection()" @click="bulkRecover">
+                    <i class="fa-solid fa-plus" aria-hidden="true" />
+                    <span v-html="trans('restore_selected')"></span>
+                    <span v-html="countSelected()"></span>
+                </button>
+
+            </div>
+        </div>
+
+        <div class="text-center text-4xl p-4" v-if="loading">
+            <i class="fa-solid fa-spinner fa-spin" aria-label="Loading" />
+        </div>
+        <div class="flex flex-col gap-4" v-else>
+            <div class="flex gap-2 flex-row">
+                <div :class="gridClass()">
+                    <Element
+                        v-for="file in files"
+                        :file="file"
+                        :isBulking="isBulking"
+                        :i18n="i18n"
+                        @recover="recoverElement(file)"
+                    >
+                    </element>
+                </div>
+            </div>
+        </div>
+    </div>
+
+    <dialog ref="newDialog" class="dialog rounded-2xl text-center" v-if="initiated">
+        <header class="bg-base-200 sm:rounded-t">
+            <h4 v-html="trans('premium_title')"></h4>
+            <button type="button" class="text-base-content" @click="closeModal(newDialog)" title="Close">
+                <i class="fa-regular fa-circle-xmark" aria-hidden="true"></i>
+                <span class="sr-only">Close</span>
+            </button>
+        </header>
+        <article class="max-w-4xl flex flex-col gap-2 text-left">
+            <div class="flex flex-col gap-1 w-full">
+                <label v-html="trans('premium')"></label>
+            </div>
+        </article>
+        <footer class="bg-base-200 p-2">
+            <menu class="">
+                <a :href="upgradeLink" v-html="trans('upgrade')" class="btn2 btn-default"></a>
+            </menu>
+        </footer>
+    </dialog>
+</template>
+
+<script setup lang="ts">
+
+import {onMounted, ref} from "vue";
+import Element from "./Element.vue";
+
+const props = defineProps<{
+    api: String
+}>()
+
+const initiated = ref(false)
+const loading = ref(false)
+const isBulking = ref(false)
+const premium = ref(false)
+const currentFile = ref()
```

Comment:
There is a lot of stuff from the gallery here that isn't used. Please clean up the file.